### PR TITLE
get all Identities through GetIdentities FCP API

### PR DESCRIPTION
**Pull request by Olezha**
https://freenet.mantishub.io/view.php?id=6490
it is necessary to check why the responses GetIdentitiesByScore and GetIdentities have a different structure:
```
for(int count = 0;; count++) {
  String identityIdFromGetIdentitiesByScore = response.sfs.get("Identity" + count);
  String identityIdFromGetIdentities = response.sfs.get("Identities." + count + ".ID");
```

### CODE REVIEW

File: src/org/freenetproject/freemail/wot/WoTConnectionImpl.java [hunk 1]
```diff
@@ -88,56 +88,25 @@
...
+		sfs.putOverwrite("Context", "");
 
 		Message response = sendBlocking(new Message(sfs, null), "Identities");
-		if(response == null) {
-			return null;
-		}
-		if(!"Identities".equals(response.sfs.get("Message"))) {
-			return null;
-		}
+		if(response == null) return null;
```
Comment (Bombe) [inline src/org/freenetproject/freemail/wot/WoTConnectionImpl.java]: Please don’t contract this into one line.

File: src/org/freenetproject/freemail/wot/WoTConnectionImpl.java [hunk 2]
```diff
@@ -88,56 +88,25 @@
...
+		if(!"Identities".equals(response.sfs.get("Message"))) return null;
```
Comment (Bombe) [inline src/org/freenetproject/freemail/wot/WoTConnectionImpl.java]: You can create a single `if` with two conditions, though.

File: src/org/freenetproject/freemail/wot/WoTConnectionImpl.java [hunk 3]
```diff
@@ -88,56 +88,25 @@
...
-		sfs.put("WantTrustValues", false);
-
-		/*
-		 * The Freemail context wasn't added prior to 0.2.1, so for a while after 0.2.1 we need to
-		 * support recipients without the Freemail context.
-		 * FIXME: Restrict this to Freemail context a few months after 0.2.1 has become mandatory
-		 */
-		sfs.putOverwrite("Context", "Freemail");
+		sfs.putOverwrite("Message", "GetIdentities");
+		sfs.putOverwrite("Context", "");
```
Comment (Bombe) [inline src/org/freenetproject/freemail/wot/WoTConnectionImpl.java]: Is it necessary to set `Context`to an empty string?

File: src/org/freenetproject/freemail/wot/IdentityMatcher.java [hunk 4]
```diff
@@ -38,23 +38,23 @@ public IdentityMatcher(WoTConnection wotConnection) {
 		this.wotConnection = wotConnection;
 	}
 
-	public Map<String, List<Identity>> matchIdentities(Set<String> recipients, String wotOwnIdentity, EnumSet<MatchMethod> methods) throws PluginNotFoundException {
-		Set<Identity> trustedIds = wotConnection.getAllTrustedIdentities(wotOwnIdentity);
-		Set<Identity> untrustedIds = wotConnection.getAllUntrustedIdentities(wotOwnIdentity);
+	public Map<String, List<Identity>> matchIdentities(
+		 Set<String> recipients, String wotOwnIdentity, EnumSet<MatchMethod> methods)
```
Comment (Bombe) [inline src/org/freenetproject/freemail/wot/IdentityMatcher.java]: Indentation with a tab and a space? That doesn’t look correct (or good). :)

File: src/org/freenetproject/freemail/wot/WoTConnectionImpl.java (hunk 2 above)
Comment (Bombe) [reply src/org/freenetproject/freemail/wot/WoTConnectionImpl.java]: If that’s too unreadable, extract it into a method with a nice, readable name. :)

File: src/org/freenetproject/freemail/wot/IdentityMatcher.java (hunk 4 above)
Comment (Olezha) [reply src/org/freenetproject/freemail/wot/IdentityMatcher.java]: This is a long line that I split to make it easier to read.

File: src/org/freenetproject/freemail/wot/WoTConnectionImpl.java (hunk 3 above)
Comment (Olezha) [reply src/org/freenetproject/freemail/wot/WoTConnectionImpl.java]: No it isn't. It is my mistake. This functionality works.

**Comment by ArneBab** [comment]
merged — thank you!

**Pull request merged**